raster-foundry/raster-foundry
# Add SceneToLayerDao project store evidence

## Overview

This PR adds projectstore evidence to the SceneToLayerDao, moves mosaic fetches for projects from projects to layers, and implements the mosaic TMS endpoint for layers so that testing instructions have anything useful to say.

### Checklist


## Notes

~[wip] until I figure out what to do about histograms and depends on #4533~ Figured it out! There are only layers -- the defualt `/project id/z/x/y` match has actually been removed from the mosaic service, and requests to it get rewritten in [a new middleware](https://github.com/raster-foundry/raster-foundry/blob/feature/js/add-scenetolayerdao-project-store/app-backend/backsplash-server/src/main/scala/com/rasterfoundry/backsplash/ProjectToProjectLayerMiddleware.scala).

## Testing Instructions

 * **run migrations** or you might spend a few minutes confused about why you don't get any imagery for anything :man_facepalming: 
 * assemble your servers
 * start your servers
 * view a project
 * copy the tms request for one of the requests that returned some data (use response sizes to pick one)
 * find the default layer id for that project
 * make the same tms request with `/layers/layerID` before the z/x/y part
 * confirm that they match

Closes #4517 

Closes #4533 

Helps with #4518
